### ISSUE TITLE
Modify DelineateIt to allow watershed delineation from all pour points

### ISSUE THREAD

**Issue by richpsharp** [natcap/invest#147]
~~RouteDEM~~ DelineateIt can create watersheds from a given set of polygons, but this feature would work directly off a DEM and create watersheds for the entire dataset, seeding a watershed at every pour point on the edge of the dem or internal edges against nodata values.

**Comment by richpsharp** [comment]
From Stacie: The main nice-to-have thing is letting the user somehow define the rough size (or minimum size) of sub-watersheds to make. Wang and Liu creates wall to wall watersheds, but you can't define anything about them. ArcSWAT at least used to (haven't tried it lately) set the rough size too, which is super helpful.

**Comment by phargogh** [comment]
If we don't get to this before the 3.9 release, we'll bump this to 3.10.

**Comment by phargogh** [comment]
@emlys we already talked about this in person a bit, but let's hold off on the subwatersheds portion of this issue for the time being, as the issue itself isn't sufficiently well-defined.

Pour point detection should still be good to go.

**Comment by emlys** [comment]
The pour point detection feature of this issue is now complete, but still need to address the subwatersheds feature before we can close this.

**Comment by phargogh** [comment]
Let's separate the two issues so as not to hold up the release.  I've created #349 for this purpose and will fill in the details of the issue once we've ironed them out.
